File: Lab1/Bill.py
class Bill:
    def __init__(self):
        self.__product_quantity: dict[str, int] = {}
        self.__total: float = 0

    def add_transaction(self, name, cost, quantity=1):
        if name in self.__product_quantity.keys():
            self.__product_quantity[name] += quantity
        else:
            self.__product_quantity[name] = quantity

        self.__total += cost*quantity

    def get_total(self):
        return self.__total

    def get_products_sold(self):
        return self.__product_quantity

File: Lab 1/src/main/java/Bill.java
import java.util.*;

public class Bill {
    private final Map<String, Integer> productQuantity = new HashMap<>();
    private double total = 0;

    public void addTransaction(String product, double price, int quantity) {
        productQuantity.put(product, productQuantity.getOrDefault(product, 0) + quantity);
        total += price * quantity;
    }

    public double getTotal() {
        return total;
    }

    public Map<String, Integer> getProductsSold() {
        return Collections.unmodifiableMap(productQuantity);
    }
}
